### PR TITLE
Data chart cleanup

### PR DESCRIPTION
**Pull request by zoemccormick**
Corresponding PR to #511 with data chart

Changes

- Use a different proxy kubernetes label name. Instead of app, we're using “gm.proxy”
- Update proxy port to use 10808
- Use snake case
- Add comments to all values files
- Expose all envvars in values files, remove all possible envvars from template files
- Clean up all global values

To test:

1. Install secrets

```bash
helm dep up secrets
helm install secrets secrets -f credentials.yaml
```

2. Install the data chart

```bash
helm dep up data
helm install data data --set=global.environment=kubernetes
```

Make sure all of the pods come up and are configured as desired

3. Check out the values files comments https://github.com/DecipherNow/helm-charts/blob/data-values/data/values.yaml

### CODE REVIEW

File: data/gm-data/templates/_envvars.tpl [hunk 1]
```diff
@@ -45,8 +45,8 @@ We use indentation in the template for readability, but the template returns the
 Most users should use the `indent` or `nindent` functions to automatically indent the proper amount. */}}
 {{- define "greymatter.envvars" }}
   {{- $top := . }}
-  {{- if .Values.global.sidecar }}
-    {{- range $name, $envvar := .Values.global.sidecar.envvars }}
+  {{- if .Values.sidecar.envvars }}
```
Comment (kaitmore) [inline data/gm-data/templates/_envvars.tpl]: https://github.com/DecipherNow/helm-charts/issues/513

File: data/gm-data/values.yaml [hunk 2]
```diff
@@ -1,118 +1,57 @@
...
-  base_path: /services/{{ $.Values.data.name | lower }}/{{ $.Values.data.version | trunc 3 }}
   image: docker.production.deciphernow.com/deciphernow/gm-data:{{ $.Values.data.version }}
-  imagePullSecret: docker.secret
-  certs_mount_point: /certs
-  imagePullPolicy: IfNotPresent
-  use_tls: true
+  image_pull_secret: '{{ .Values.global.image_pull_secret }}'
+  image_pull_policy: IfNotPresent
   master_key: ac8923[lkn43589vi23kl4rfgv0ws
+  # Location and size of pvc to persist data if not using s3
```
Comment (moosecanswim) [inline data/gm-data/values.yaml]: Looks like size was removed

File: data/gm-data/values.yaml [hunk 3]
```diff
@@ -1,118 +1,57 @@
...
-    use_provided_certs: true
-
+  # Port for Grey Matter Control. Used in sidecar envvars
+  control_port: 50000
+  # Zone for Grey Matter objects. Used in sidecar envvars
+  zone: zone-default-zone
+  # Whether or not to use spire
   spire:
     enabled: false
+  # Whether to register services with Consul and where to find the Consul server
```
Comment (moosecanswim) [inline data/gm-data/values.yaml]: Should there be another key that will point to the consul server?

File: data/gm-data/values.yaml [hunk 4]
```diff
@@ -279,37 +214,59 @@ mongo:
     mongo_cn:
       type: value
       value: 'localhost'
-  
+    # If true, mongo.secret must be set, and mongo_key, mongo_trust, and mongo_cert must be specified as below
```
Comment (moosecanswim) [inline data/gm-data/values.yaml]: Unsure where `mongo.secret` is set

File: data/gm-data/values.yaml [hunk 5]
```diff
@@ -279,37 +214,59 @@ mongo:
...
     #  ca: ca_b64
     #  cert: cert_b64
     #  key: key_b64
 
+# Sets default sidecar configurations for all sidecars
 sidecar:
-  version: '{{- $.Values.global.data.sidecar.version | default $.Values.global.sidecar.version }}'
-  image: 'docker-dev.production.deciphernow.com/deciphernow/gm-proxy:{{ tpl $.Values.sidecar.version $ }}'
-  imagePullPolicy: IfNotPresent
+  version: 'latest'
```
Comment (moosecanswim) [inline data/gm-data/values.yaml]: We should pin this to a version number

**Comment by kaitmore** [comment]
Installing the data chart works great! I ran into a few issues trying to install individual charts:

```
$ helm install data data/gm-data --set=global.environment=kubernetes

Error: template: gm-data/templates/data.yaml:58:18: executing "gm-data/templates/data.yaml" at <tpl .Values.sidecar.image $>: error calling tpl: error during tpl function execution for "docker-dev.production.deciphernow.com/deciphernow/gm-proxy:{{ tpl $.Values.sidecar.version $ }}": template: gm-data/templates/data.yaml:1:62: executing "gm-data/templates/data.yaml" at <tpl $.Values.sidecar.version $>: error calling tpl: error during tpl function execution for "{{ .Values.global.sidecar.version | default \"latest\" }}": template: gm-data/templates/data.yaml:1:10: executing "gm-data/templates/data.yaml" at <.Values.global.sidecar.version>: nil pointer evaluating interface {}.version
```

```
$ helm install data data/jwt --set=global.environment=kubernetes

Error: template: jwt/templates/jwt.yaml:35:12: executing "jwt/templates/jwt.yaml" at <include "jwt.envvars" (dict "envvar" .Values.jwt.envvars "top" $)>: error calling include: template: jwt/templates/_envvars.tpl:16:23: executing "jwt.envvars" at <$envvar.value>: wrong type for value; expected string; got bool
```

```
$ helm install data data/jwt-gov --set=global.environment=kubernetes

Error: template: jwt-gov/templates/jwt.yaml:68:12: executing "jwt-gov/templates/jwt.yaml" at <include "sidecar.envvars" .>: error calling include: template: jwt-gov/templates/_envvars.tpl:27:20: executing "sidecar.envvars" at <.Values.global.sidecar.envvars>: nil pointer evaluating interface {}.envvars
```

File: data/values.yaml [hunk 6]
```diff
@@ -1,137 +1,69 @@
...
-      # This value must match global.control_api_tls
-      egressTLS: true
+  # The port where the edge proxy is running, needed if consul is enabled
+  edge_port: 10808
+  # Port for Grey Matter Control. Used in sidecar envvars
+  control_port: 50000
 
-  sidecar:
-    version: latest
+  # Overrides the default configurations for all sidecars in all subcharts
```
Comment (kaitmore) [inline data/values.yaml]: Could we modify this to say "Default configuration for all sidecars in all subcharts"

**Comment by moosecanswim** [comment]
@kaitmore were you able to start up jwt-gov standalone?  Im getting a `/bin/sh: /app/gm-jwt-security-gov.linux: not found` assuming the docker file did not pick up the binary correctly

**Comment by kaitmore** [comment]
@moosecanswim I didn't get any errors on install, but I also didn't have auth for that jwt-gov image so I couldn't get as far as you.

**Comment by moosecanswim** [comment]
@kaitmore gotcha.  I'll check in on the image to see whats up.

**Comment by moosecanswim** [comment]
@zoemccormick @kaitmore  I opened an issue against the jwt gov repo (https://github.com/DecipherNow/gm-jwt-security-gov/issues/126)

**Comment by zoemccormick** [comment]
@moosecanswim  I saw the same issue - I don't know anything about jwt-gov so I don't know what's safe to change/take out. I figured I would file an issue to fix its args

**Comment by moosecanswim** [comment]
@zoemccormick you should be able to change the command from `/app/gm-jwt-security-gov` to `/gm-jwt-security-gov`

**Comment by moosecanswim** [comment]
@zoemccormick the arguments should be 
```
args:
    [
      '-c',
      'if [[ ! -d ./certs ]]; then mkdir -p ./certs; fi && if [[ ! -d ./certs/aac ]]; then mkdir -p ./certs/aac; fi && if [[ ! -d ./etc ]]; then mkdir -p ./certs; fi && /gm-jwt-security-gov/gm-jwt-security-gov',

    ]
```

**Pull request merged**